plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -75,90 +84,222 @@ protected DistributedQueryRunner createQueryRunner()
-        private Predicate<String> pathPredicate = path -> true;
-        private Optional<TrackingFileIoProvider.OperationType> type = Optional.empty();
+        assertUpdate("CREATE TABLE test_show_stats AS SELECT 2 as age", 1);
+
+        assertFileSystemAccesses("SHOW STATS FOR test_show_stats",
+                ImmutableMultiset.builder()
+                        .add(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH))
+                        .add(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM))
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 2)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 2)
+                        .build());
+    }
 
-        public OperationCounts(Map<OperationContext, Integer> allCounts)
+    @Test
+    public void testShowStatsForTableWithFilter()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_show_stats_with_filter AS SELECT 2 as age", 1);
+
+        assertFileSystemAccesses("SHOW STATS FOR (SELECT * FROM test_show_stats_with_filter where age >= 2)",
+                ImmutableMultiset.builder()
+                        .add(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH))
+                        .add(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM))
+                        .add(new FileOperation(METADATA, INPUT_FILE_NEW_STREAM))
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 2)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 2)
+                        .build());
+    }
+
+    private void assertFileSystemAccesses(String query, Multiset expectedAccesses)
+            throws Exception
+    {
+        resetCounts();
+        getDistributedQueryRunner().executeWithQueryId(TEST_SESSION, query);
+        assertThat(getOperations()).containsExactlyInAnyOrderElementsOf(expectedAccesses);
+    }
+
+    private static void resetCounts()
+            throws Exception
+    {
+        BEAN_SERVER.invoke(new ObjectName(BEAN_NAME), OPERATION_RESET, null, null);
+    }
+
+    private static Multiset<FileOperation> getOperations()
+            throws Exception
+    {
+        return ((Map<OperationContext, Integer>) BEAN_SERVER.getAttribute(new ObjectName(BEAN_NAME), OPERATION_COUNTS_ATTRIBUTE))
+                .entrySet().stream()
+                .map(entry -> entry.getKey())
+                .map(FileOperation::new)
+                .collect(Collectors.toCollection(HashMultiset::create));
+    }
+
+    static class FileOperation
+    {
+        private final FileType fileType;
+        private final OperationType operationType;
+
+        public FileOperation(FileType fileType, OperationType operationType)
         {
-            this.allCounts = requireNonNull(allCounts, "allCounts is null");
+            this.fileType = fileType;
```

Comment:
Check for null

Reply:
@homar you may find https://gist.github.com/findepi/b597d2bc91e5a30c2a44fd65fe252f0b useful